### PR TITLE
calo debug data packet description

### PR DESCRIPTION
**Pull request by gianipez**

### CODE REVIEW

**Comment by giro94** [comment]
**CalorimeterTestDataDecoder** can now properly read calo debug hits with the usual function `GetCalorimeterHitData()`.
Pair of 16-bit words are swapped prior to 16-to-12 bit conversion.
The process should be memory efficient.

Checks for begin word (0xAAA) and last sample word (0xFFF) are in place.
Other checks are to be implemented.

This class will be merged with the previous **CalorimeterDataDecoder** and adapted to read standard calo hits and debug counters with dedicated functions in a future commit.

**Comment by giro94** [comment]
CalorimeterTestDataDecoder class is now merged into **CalorimeterDataDecoder**.
It can decode both standard and debug hits (as CalorimeterHitDataPacket or CalorimeterHitTestDataPacket).
Didn't test the standard hit decoding on real/artificial data yet.

TODOs:
- decoding counters data
- Compatibility issues: used "ChannelNumber" in standard hit packet for backward compatibility. Also, no bit-fields on post-waveform words.

File: artdaq-core-mu2e/Data/CalorimeterDataDecoder.hh [hunk 1]
```diff
@@ -3,53 +3,90 @@
...
+        }
+    };
 
-    // CalorimeterHitDataPacket: Each hit is readout as a variable length sequence of data packets
     struct CalorimeterHitDataPacket
     {
       uint16_t DetectorType : 3; // subdetector type e.g. CALO=0, CAPHRI = 1, TRAD = 2, LASER = 3
       uint16_t BoardID : 8; //unique board ID from 0 - 255
-      uint16_t ChannelNumber : 6; // channel ID from 0-19
+      uint16_t ChannelNumber : 5; // channel ID from 0-19
```
Comment (gianipez) [inline artdaq-core-mu2e/Data/CalorimeterDataDecoder.hh:59]: Why has this value changed?

Comment (giro94) [reply artdaq-core-mu2e/Data/CalorimeterDataDecoder.hh:59]: I changed it to match [DocDB@4914](https://mu2e-docdb.fnal.gov/cgi-bin/sso/ShowDocument?docid=4914):
image

**Comment by giro94** [comment]
Restored original GetCalorimeterHitData() and GetCalorimeterHitDataForTrigger() for the standard calo hits.
I commented my new version.

**Comment by giro94** [comment]
Restored original CalorimeterHitDataPacket as well.

**Comment by gianipez** [comment]
Paolo made tests showing that the changes introduced allow to read data acquired during the GR3. 
Merging

**Pull request merged**